topology.go

```diff
@@ -23,8 +72,167 @@ type TopologyNode struct {
 	Address     string   `xml:"address,attr" json:"address"`
 	Port        uint16   `xml:"port,attr" json:"port"`
 	APIPort     uint16   `xml:"apiport,attr" json:"apiport"`
-	Weight      int      `xml:"weight,attr" json:"weight"`
-	NumberNodes int      `xml:"-" json:"n"`
+	Weight      uint16   `xml:"weight,attr" json:"weight"`
+	Side        TopoSide `xml:"side,attr", json:"side"`
+	WriteCopies uint8    `xml:"-" json:"n"`
+}
+
+func (topo *Topology) compile() error {
+	nslots := 0
+	if topo.WriteCopies == 0 {
+		topo.WriteCopies = topo.OldWriteCopies
+	}
+	topo.OldWriteCopies = topo.WriteCopies
+	for _, node := range topo.Nodes {
+		node.ID = strings.ToLower(node.ID)
+		if node.Side != 0 {
+			topo.use_side = true
+		}
+		nslots += int(node.Weight)
+	}
+	hash := sha256.New()
+	for _, node := range topo.Nodes {
+		hash.Write([]byte(node.ID))
```

Comment:
These calls to hash.Write do not check the error that hash.Write can return.

Reply:
AFAICT that function cannot throw an error.